hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoUpdateMeta.java

```diff
@@ -0,0 +1,189 @@
+package com.hedera.services.usage.crypto;
+
+/*-
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hederahashgraph.api.proto.java.CryptoUpdateTransactionBody;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_ENTITY_ID_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.INT_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.LONG_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.getAccountKeyStorageSize;
+
+public class CryptoUpdateMeta {
+
+	private final long keyBytesUsed;
+	private final long msgBytesUsed;
+	private final long memoSize;
+	private final long effectiveNow;
+	private final long expiry;
+	private final boolean hasProxy;
+	private final int maxAutomaticAssociations;
+	private final boolean hasMaxAutomaticAssociations;
+
+	public CryptoUpdateMeta(Builder builder) {
+		keyBytesUsed = builder.keyBytesUsed;
+		msgBytesUsed = builder.msgBytesUsed;
```

Comment:
Why do we need this here? There doesn't seem to have corresponding field in `CryptoUpdateTransactionBody` or in the context.

Reply:
Its a place holder for the calculation we do here `private int bytesUsedInTxn(CryptoUpdateTransactionBody op)`. And Builder class mostly for testing purposes.

Reply:
Hmm, I thought `CryptoUpdateMeta` is used also in real world's fee calculation as well.

Reply:
It is. I though you were asking about `Builder` class in `CryptoUpdateMeta`